### ISSUE TITLE
1.7.3 crashed on setup of upgrade_schema_test.py

### ISSUE THREAD

**Issue by abvgedeika** [scylladb/scylla#2653]
*Installation details*
Scylla version (or git commit hash):      1.7.3: ami_id_db_scylla: 'ami-9b5a4c8d'
Cluster size: 3
OS (RHEL/CentOS/Ubuntu/AWS AMI): AWS AMI

test: https://github.com/abvgedeika/scylla-cluster-tests/blob/test_upgrade_cql_queries/upgrade_schema_test.py

I'm not able to get stacktrace because
```
[root@ip-172-30-0-38 centos]# /opt/scylladb/bin/gdb /usr/bin/scylla /var/lib/scylla/coredump/core.scylla.994.7b48ae0a4daf42759d4e28812f7be2c2.4
core.scylla.994.7b48ae0a4daf42759d4e28812f7be2c2.4452.1501534786000000.gz  core.scylla.994.7b48ae0a4daf42759d4e28812f7be2c2.4808.1501535021000000
[root@ip-172-30-0-38 centos]# /opt/scylladb/bin/gdb /usr/bin/scylla /var/lib/scylla/coredump/core.scylla.994.7b48ae0a4daf42759d4e28812f7be2c2.4808.1501535021000000 
GNU gdb (GDB) Red Hat Enterprise Linux 7.10.1-30.el7.centos
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/bin/scylla...Reading symbols from /usr/lib/debug/usr/bin/scylla.debug...../../gdb/cp-support.c:1595: demangler-warning: unable to demangle '_ZN8futurizeI13frozen_schemaE5applyIRZN7seastar7shardedIN7service13storage_proxyEE9invoke_onIZZNS6_22init_messaging_serviceEvENKUljN5utils4UUIDEE6_clEjSA_EUlOT_E_6futureIJS0_EEEET0_jSD_EUlvE_JEEESG_SD_DpOT0_' (demangler failed with signal 11)
Unable to dump core, use `ulimit -c unlimited' before executing GDB next time.
../../gdb/cp-support.c:1608: demangler-warning: unable to demangle '_ZN8futurizeI13frozen_schemaE5applyIRZN7seastar7shardedIN7service13storage_proxyEE9invoke_onIZZNS6_22init_messaging_serviceEvENKUljN5utils4UUIDEE6_clEjSA_EUlOT_E_6futureIJS0_EEEET0_jSD_EUlvE_JEEESG_SD_DpOT0_' (demangler failed with signal 11)
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Quit this debugging session? (y or n) y
```

```
ul 31 21:02:18 ip-172-30-0-38 scylla[4742]:  [shard 4] database - Setting compaction strategy of test_upgrade_schema_ks.test9 to SizeTieredCompactionStrategy
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]:  [shard 3] database - Setting compaction strategy of test_upgrade_schema_ks.test1 to SizeTieredCompactionStrategy
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: scylla: schema.cc:800: sstring cell_comparator::to_sstring(const schema&): Assertion `s.is_dense()' failed.
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: Aborting on shard 2.
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: Backtrace:
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00000000004e06ca
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00000000004e07d3
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00007fe6ab4c80ff
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00007fe6ab12c5f6
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00007fe6ab12dce7
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00007fe6ab125565
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00007fe6ab125611
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00000000009dd552
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x0000000001022ace
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x0000000001035350
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00000000009e87e8
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00000000009ec7cc
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x000000000091a00a
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x000000000095531e
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x000000000095aa27
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x000000000095fe46
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00000000004e213f
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x0000000000531c32
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x0000000000533470
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x000000000058ed9d
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00007fe6ab4c0dc4
Jul 31 21:02:18 ip-172-30-0-38 scylla[4742]: 0x00007fe6ab1edcec
Jul 31 21:02:18 ip-172-30-0-38 systemd-coredump[4767]: Removed old coredump core.scylla.994.7b48ae0a4daf42759d4e28812f7be2c2.4452.1501534786000000.
Jul 31 21:02:32 ip-172-30-0-38 systemd-coredump[4628]: Process 4603 (scylla) of user 994 dumped core.
                                                       
                                                       Stack trace of thread 4605:
                                                       #0  0x00007f15205b55f7 raise (libc.so.6)
                                                       #1  0x00007f15205b6e28 abort (libc.so.6)
                                                       #2  0x00007f15205ae566 __assert_fail_base (libc.so.6)
                                                       #3  0x00007f15205ae612 __assert_fail (libc.so.6)
                                                       #4  0x00000000009dd553 _ZN15cell_comparator10to_sstringERK6schema (scylla)
                                                       #5  0x0000000001022acf make_table_mutations (scylla)
                                                       #6  0x0000000001035351 _ZN2db13schema_tables21make_schema_mutationsE13lw_shared_ptrIK6schemaElb (scylla)
                                                       #7  0x00000000009e87e9 operator() (scylla)
                                                       #8  0x00000000009ec7cd _ZN15schema_registry5learnERK13lw_shared_ptrIK6schemaE (scylla)
                                                       #9  0x000000000091a00b _ZN8database17add_column_familyER8keyspace13lw_shared_ptrIK6schemaEN13column_family6configE (scylla)
                                                       #10 0x000000000095531f _ZN8database36add_column_family_and_make_directoryE13lw_shared_ptrIK6schemaE (scylla)
                                                       #11 0x000000000095aa28 operator()<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > (scylla)
                                                       #12 0x000000000095fe47 operator()<future_state<std::map<basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema>, std::less<basic_sstring<char, unsigned int, 15u> >, std::allocator<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > > > > (scylla)
                                                       #13 0x00000000004e2140 _ZN7reactor9run_tasksER15circular_bufferISt10unique_ptrI4taskSt14default_deleteIS2_EESaIS5_EE (scylla)
                                                       #14 0x0000000000531c33 _ZN7reactor3runEv (scylla)
                                                       #15 0x0000000000533471 _ZZN3smp9configureEN5boost15program_options13variables_mapEENKUlvE1_clEv.constprop.4081 (scylla)
                                                       #16 0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #17 0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #18 0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4612:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e0825 _ZN9file_desc4readEPvm (scylla)
                                                       #2  0x00000000019d6a80 execute_native_thread_routine (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4608:
                                                       #0  0x00007f1520676f27 epoll_pwait (libc.so.6)
                                                       #1  0x00000000004f647b _ZN21reactor_backend_epoll16wait_and_processEiPK10__sigset_t (scylla)
                                                       #2  0x00000000004f65f1 _ZN7reactor16wait_and_processEiPK10__sigset_t (scylla)
                                                       #3  0x000000000053214b _ZN7reactor3runEv (scylla)
                                                       #4  0x0000000000533471 _ZZN3smp9configureEN5boost15program_options13variables_mapEENKUlvE1_clEv.constprop.4081 (scylla)
                                                       #5  0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #6  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #7  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4616:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e0825 _ZN9file_desc4readEPvm (scylla)
                                                       #2  0x00000000019d6a80 execute_native_thread_routine (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4621:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e0825 _ZN9file_desc4readEPvm (scylla)
                                                       #2  0x00000000019d6a80 execute_native_thread_routine (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4620:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e0825 _ZN9file_desc4readEPvm (scylla)
                                                       #2  0x00000000019d6a80 execute_native_thread_routine (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4614:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e0825 _ZN9file_desc4readEPvm (scylla)
                                                       #2  0x00000000019d6a80 execute_native_thread_routine (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4619:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e0825 _ZN9file_desc4readEPvm (scylla)
                                                       #2  0x00000000019d6a80 execute_native_thread_routine (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4618:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e0825 _ZN9file_desc4readEPvm (scylla)
                                                       #2  0x00000000019d6a80 execute_native_thread_routine (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4613:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e2a06 _ZN11thread_pool4workEv (scylla)
                                                       #2  0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4617:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e0825 _ZN9file_desc4readEPvm (scylla)
                                                       #2  0x00000000019d6a80 execute_native_thread_routine (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4626:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e2a06 _ZN11thread_pool4workEv (scylla)
                                                       #2  0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4624:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e2a06 _ZN11thread_pool4workEv (scylla)
                                                       #2  0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4622:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e2a06 _ZN11thread_pool4workEv (scylla)
                                                       #2  0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4615:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e2a06 _ZN11thread_pool4workEv (scylla)
                                                       #2  0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4625:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e2a06 _ZN11thread_pool4workEv (scylla)
                                                       #2  0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4627:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e2a06 _ZN11thread_pool4workEv (scylla)
                                                       #2  0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4623:
                                                       #0  0x00007f152095022d read (libpthread.so.0)
                                                       #1  0x00000000004e2a06 _ZN11thread_pool4workEv (scylla)
                                                       #2  0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #3  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #4  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4609:
                                                       #0  0x00007ffd1c23b701 __vdso_clock_gettime (linux-vdso.so.1)
                                                       #1  0x00007f152068a93d __clock_gettime (libc.so.6)
                                                       #2  0x00000000019d17a1 _ZNSt6chrono3_V212steady_clock3nowEv (scylla)
                                                       #3  0x0000000000980758 _ZN5utils15latency_counter3nowEv (scylla)
                                                       #4  0x00000000008d2b94 _ZN5utils39timed_rate_moving_average_and_histogramC4Eml (scylla)
                                                       #5  0x000000000091a107 _ZN13column_familyC4E13lw_shared_ptrIK6schemaENS_6configERN2db9commitlogER18compaction_manager (scylla)
                                                       #6  0x000000000095531f _ZN8database36add_column_family_and_make_directoryE13lw_shared_ptrIK6schemaE (scylla)
                                                       #7  0x000000000095aa28 operator()<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > (scylla)
                                                       #8  0x000000000095fe47 operator()<future_state<std::map<basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema>, std::less<basic_sstring<char, unsigned int, 15u> >, std::allocator<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > > > > (scylla)
                                                       #9  0x00000000004e2140 _ZN7reactor9run_tasksER15circular_bufferISt10unique_ptrI4taskSt14default_deleteIS2_EESaIS5_EE (scylla)
                                                       #10 0x0000000000531c33 _ZN7reactor3runEv (scylla)
                                                       #11 0x0000000000533471 _ZZN3smp9configureEN5boost15program_options13variables_mapEENKUlvE1_clEv.constprop.4081 (scylla)
                                                       #12 0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #13 0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #14 0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4611:
                                                       #0  0x00007f15206d0573 __memmove_ssse3_back (libc.so.6)
                                                       #1  0x0000000000548000 _ZNSt11__copy_moveILb0ELb1ESt26random_access_iterator_tagE8__copy_mIcEEPT_PKS3_S6_S4_ (scylla)
                                                       #2  0x0000000000fc9de3 _ZN13basic_sstringIcjLj15EEC4EPKc (scylla)
                                                       #3  0x0000000000ff1651 _ZN2db13schema_tables29create_column_from_column_rowERKN5query14result_set_rowE13basic_sstringIcjLj15EES6_b (scylla)
                                                       #4  0x0000000000ff1df3 _ZN2db13schema_tables31create_columns_from_column_rowsERKN5query10result_setERK13basic_sstringIcjLj15EES8_b (scylla)
                                                       #5  0x0000000001013630 _ZN2db13schema_tables27create_table_from_mutationsE16schema_mutationsNSt12experimental15fundamentals_v18optionalIN5utils4UUIDEEE (scylla)
                                                       #6  0x00000000009e8177 _ZNK13frozen_schema8unfreezeEv (scylla)
                                                       #7  0x00000000009ec1c7 _ZN21schema_registry_entry10get_schemaEv (scylla)
                                                       #8  0x00000000009ec38e _ZN21schema_registry_entry4loadE13frozen_schema (scylla)
                                                       #9  0x00000000009ec7e4 _ZN15schema_registry5learnERK13lw_shared_ptrIK6schemaE (scylla)
                                                       #10 0x000000000091a00b _ZN8database17add_column_familyER8keyspace13lw_shared_ptrIK6schemaEN13column_family6configE (scylla)
                                                       #11 0x000000000095531f _ZN8database36add_column_family_and_make_directoryE13lw_shared_ptrIK6schemaE (scylla)
                                                       #12 0x000000000095aa28 operator()<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > (scylla)
                                                       #13 0x000000000095fe47 operator()<future_state<std::map<basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema>, std::less<basic_sstring<char, unsigned int, 15u> >, std::allocator<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > > > > (scylla)
                                                       #14 0x00000000004e2140 _ZN7reactor9run_tasksER15circular_bufferISt10unique_ptrI4taskSt14default_deleteIS2_EESaIS5_EE (scylla)
                                                       #15 0x0000000000531c33 _ZN7reactor3runEv (scylla)
                                                       #16 0x0000000000533471 _ZZN3smp9configureEN5boost15program_options13variables_mapEENKUlvE1_clEv.constprop.4081 (scylla)
                                                       #17 0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #18 0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #19 0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4607:
                                                       #0  0x00007f1520676f27 epoll_pwait (libc.so.6)
                                                       #1  0x00000000004f647b _ZN21reactor_backend_epoll16wait_and_processEiPK10__sigset_t (scylla)
                                                       #2  0x00000000004f65f1 _ZN7reactor16wait_and_processEiPK10__sigset_t (scylla)
                                                       #3  0x000000000053214b _ZN7reactor3runEv (scylla)
                                                       #4  0x0000000000533471 _ZZN3smp9configureEN5boost15program_options13variables_mapEENKUlvE1_clEv.constprop.4081 (scylla)
                                                       #5  0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #6  0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #7  0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4610:
                                                       #0  0x00007f15205c4d85 makecontext (libc.so.6)
                                                       #1  0x00000000005b654c _ZN7seastar14thread_context5setupEv (scylla)
                                                       #2  0x00000000005b6d2c _ZN7seastar14thread_contextC2ENS_17thread_attributesESt8functionIFvvEE (scylla)
                                                       #3  0x000000000092fad2 make_unique<seastar::thread_context, seastar::thread_attributes, seastar::async(seastar::thread_attributes, Func&&, Args&& ...)::<lambda(seastar::async(seastar::thread_attributes, Func&&, Args&& ...)::work&)> mutable [with Func = keyspace::make_directory_for_column_family(const sstring&, utils::UUID)::<lambda()>; Args = {}]::<lambda()>&> (scylla)
                                                       #4  0x0000000000955368 _ZN8database36add_column_family_and_make_directoryE13lw_shared_ptrIK6schemaE (scylla)
                                                       #5  0x000000000095aa28 operator()<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > (scylla)
                                                       #6  0x000000000095fe47 operator()<future_state<std::map<basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema>, std::less<basic_sstring<char, unsigned int, 15u> >, std::allocator<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > > > > (scylla)
                                                       #7  0x00000000004e2140 _ZN7reactor9run_tasksER15circular_bufferISt10unique_ptrI4taskSt14default_deleteIS2_EESaIS5_EE (scylla)
                                                       #8  0x0000000000531c33 _ZN7reactor3runEv (scylla)
                                                       #9  0x0000000000533471 _ZZN3smp9configureEN5boost15program_options13variables_mapEENKUlvE1_clEv.constprop.4081 (scylla)
                                                       #10 0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #11 0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #12 0x00007f1520676ced __clone (libc.so.6)
                                                       
                                                       Stack trace of thread 4603:
                                                       #0  0x00000000008d32e0 _ZN5boost9intrusive17slist_node_traitsIPvE8set_nextERKPNS0_10slist_nodeIS2_EES8_ (scylla)
                                                       #1  0x000000000091a107 _ZN13column_familyC4E13lw_shared_ptrIK6schemaENS_6configERN2db9commitlogER18compaction_manager (scylla)
                                                       #2  0x000000000095531f _ZN8database36add_column_family_and_make_directoryE13lw_shared_ptrIK6schemaE (scylla)
                                                       #3  0x000000000095aa28 operator()<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > (scylla)
                                                       #4  0x000000000095fe47 operator()<future_state<std::map<basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema>, std::less<basic_sstring<char, unsigned int, 15u> >, std::allocator<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > > > > (scylla)
                                                       #5  0x00000000004e2140 _ZN7reactor9run_tasksER15circular_bufferISt10unique_ptrI4taskSt14default_deleteIS2_EESaIS5_EE (scylla)
                                                       #6  0x0000000000531c33 _ZN7reactor3runEv (scylla)
                                                       #7  0x00000000005b1c3d _ZN12app_template14run_deprecatedEiPPcOSt8functionIFvvEE (scylla)
                                                       #8  0x00000000004733de main (scylla)
                                                       #9  0x00007f15205a1b15 __libc_start_main (libc.so.6)
                                                       #10 0x00000000004de0df _start (scylla)
                                                       
                                                       Stack trace of thread 4606:
                                                       #0  0x0000000000590d60 _ZN6memory10small_pool8allocateEv (scylla)
                                                       #1  0x0000000000591a0e _ZN6memory8allocateEm (scylla)
                                                       #2  0x0000000000591b15 _Znam (scylla)
                                                       #3  0x0000000000bf5fd2 _ZN22shared_ptr_make_helperIN8sstables31size_tiered_compaction_strategyELb0EE4makeIJS1_EEE10shared_ptrIS1_EDpOT_ (scylla)
                                                       #4  0x00000000008b60b2 _ZN13column_family23set_compaction_strategyEN8sstables24compaction_strategy_typeE (scylla)
                                                       #5  0x000000000091a2ee _ZN13column_family16start_compactionEv (scylla)
                                                       #6  0x000000000095531f _ZN8database36add_column_family_and_make_directoryE13lw_shared_ptrIK6schemaE (scylla)
                                                       #7  0x000000000095aa28 operator()<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > (scylla)
                                                       #8  0x000000000095fe47 operator()<future_state<std::map<basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema>, std::less<basic_sstring<char, unsigned int, 15u> >, std::allocator<std::pair<const basic_sstring<char, unsigned int, 15u>, lw_shared_ptr<const schema> > > > > > (scylla)
                                                       #9  0x00000000004e2140 _ZN7reactor9run_tasksER15circular_bufferISt10unique_ptrI4taskSt14default_deleteIS2_EESaIS5_EE (scylla)
                                                       #10 0x0000000000531c33 _ZN7reactor3runEv (scylla)
                                                       #11 0x0000000000533471 _ZZN3smp9configureEN5boost15program_options13variables_mapEENKUlvE1_clEv.constprop.4081 (scylla)
                                                       #12 0x000000000058ed9e _ZNKSt8functionIFvvEEclEv (scylla)
                                                       #13 0x00007f1520949dc5 start_thread (libpthread.so.0)
                                                       #14 0x00007f1520676ced __clone (libc.so.6)
Jul 31 21:03:38 ip-172-30-0-38 systemd[1]: scylla-server.service: main process exited, code=killed, status=6/ABRT
```
all logs
https://gist.github.com/abvgedeika/449cee034633de996394d683d5136c8d

please note that the issue reproduced in automation tests https://github.com/abvgedeika/scylla-cluster-tests/blob/test_upgrade_cql_queries/upgrade_schema_test.py
when I run the following manual steps it's not reproduced:
1. install 1.7.2/3 3 node cluster
2. run script https://gist.github.com/abvgedeika/e8d4882bf4b404de6007331a205ce04a as:
schema_upgrade_tests.py setup
schema_upgrade_tests.py validate

**Comment by abvgedeika** [comment]
jenkins run http://jenkins.cloudius-systems.com:8080/job/scylla-ami-upgrade-tests/100/label=master/console

**Comment by duarten** [comment]
@abvgedeika Which test case in particular causes this? 

Also, not that you could have just answered "no" to gdb and it should still work. Is the core dump uploaded somewhere?

**Comment by duarten** [comment]
I'm guessing it happens due to #2588 in TestThriftTableConvertedFromDenseToSparse, which is fixed by https://github.com/scylladb/scylla/commit/8d46c4e049156afd297b4fd6f99e89b6535cfb8e, but I don't think it's merged to 1.7, but is fixed in 2.0.

**Comment by duarten** [comment]
Also note that the fix is to reject such operations.

**Comment by abvgedeika** [comment]
Thanks @duarten 
'no' really works just need to wait)

and scylla crashed on TestThriftTableConvertedFromDenseToSparse, by mistake I enabled this test in jenkins/repository. so, it's known issues for 1.7 and fixed in 2.0 only
